backend/src/impl/analyses/significance_analysis.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+from explainaboard.info import SysOutputInfo
+from explainaboard.metrics.metric import MetricStats
+from explainaboard.metrics.registry import get_metric_config_class
+from explainaboard.utils.typing_utils import unwrap
+
+
+def update_metric_config(system_output: SysOutputInfo) -> SysOutputInfo:
```

Comment:
If I'm not misunderstanding, it seems that this function is necessary because deserialization of MetricConfig is not working properly when SysOutputInfo is deserialized? Maybe we should fix the upstream problem (incomplete deserialization of SysOutputInfo) instead of having a separate function to patch the problem. If I'm misunderstanding please just tell me though, I don't have 100% confidence.

If we're going to keep the function, the name "update" is not very descriptive, maybe "deserialize" or something would be better?